HRIS.Services.Tests/Services/EmployeeTypesServiceUnitTest.cs

```diff
@@ -15,57 +15,80 @@ public class EmployeeTypesServiceUnitTest
     {
         var employeeType = EmployeeTypeTestData.DeveloperType;
         var employeeTypeDto = EmployeeTypeTestData.DeveloperType.ToDto();
 
-        _dbMock.Setup(r => r.EmployeeType.Any(It.IsAny<Expression<Func<EmployeeType, bool>>>()))
-               .ReturnsAsync(false);
+        _identity.Setup(i => i.Role).Returns("Admin");
+        _identity.Setup(x => x.EmployeeId).Returns(1);
+        _employeeTypeServiceMock.Setup(x => x.EmployeeTypeExists(employeeTypeDto.Id)).ReturnsAsync(true);
+
+        _employeeTypeServiceMock.Setup(x => x.CreateEmployeeType(employeeTypeDto)).ReturnsAsync(employeeTypeDto);
 
-        _employeeTypeServiceMock.Setup(r => r.GetEmployeeType(It.IsAny<string>())).Throws(new Exception());
-        _employeeTypeServiceMock.Setup(x => x.SaveEmployeeType(employeeTypeDto)).ReturnsAsync(employeeTypeDto);
-        _employeeTypeServiceMock.Setup(x => x.GetEmployeeType(employeeType.Name!)).ReturnsAsync(employeeTypeDto);
+        _employeeTypeServiceMock.Setup(x => x.GetEmployeeTypeByName(employeeType.Name!)).ReturnsAsync(employeeTypeDto);
 
         _dbMock.Setup(r => r.EmployeeType.Add(It.IsAny<EmployeeType>())).ReturnsAsync(employeeType);
         _dbMock.Setup(x => x.EmployeeType.Add(employeeType)).ReturnsAsync(employeeType);
 
-        var result = await employeeTypeService.SaveEmployeeType(employeeTypeDto);
+        var result = await _employeeTypeService.CreateEmployeeType(employeeTypeDto);
 
         Assert.NotNull(result);
-        Assert.Equivalent(employeeTypeDto, result);
         Assert.Equal(employeeTypeDto.Id, result.Id);
         Assert.Equal(employeeTypeDto.Name, result.Name);
     }
 
     [Fact]
-    public async Task SaveEmployeeTypeTestSuccessSave()
+    public async Task CreateEmployeeTypeDoesNotExists()
     {
         var employeeType = EmployeeTypeTestData.DeveloperType;
         var employeeTypeDto = EmployeeTypeTestData.DeveloperType.ToDto();
 
-        _employeeTypeServiceMock.Setup(x => x.SaveEmployeeType(employeeTypeDto)).ReturnsAsync(employeeTypeDto);
+        _dbMock.Setup(x => x.EmployeeType.Any(It.IsAny<Expression<Func<EmployeeType, bool>>>()))
+               .ReturnsAsync(true);
+
+        _dbMock.Setup(r => r.EmployeeType.Add(It.IsAny<EmployeeType>())).ReturnsAsync(employeeType);
 
-        _employeeTypeServiceMock.Setup(x => x.GetEmployeeType(employeeType.Name!)).ReturnsAsync(employeeTypeDto);
+        await Assert.ThrowsAsync<CustomException>(() => _employeeTypeService.CreateEmployeeType(employeeTypeDto));
+
+        _dbMock.Verify(x => x.EmployeeType.Add(It.IsAny<EmployeeType>()), Times.Never);
+        _dbMock.Verify(x => x.EmployeeType.FirstOrDefault(It.IsAny<Expression<Func<EmployeeType, bool>>>()), Times.Never);
+    }
+
+    [Fact]
+    public async Task CreateEmployeeTypeTestUnauthorized()
+    {
+        var employeeType = EmployeeTypeTestData.DeveloperType;
+        var employeeTypeDto = EmployeeTypeTestData.DeveloperType.ToDto();
```

Comment:
Refactor: We don't need to repeat these two lines of code in all the unit tests. We can set it once in the constructor and then use the variables where needed. Double check all the unit tests do not repeat code.

For example:
 
 public EmployeeTypesServiceUnitTest()
    {
    ...
    _employeeType = EmployeeTypeTestData.DeveloperType;
    _emloyeeTypeDto = EmployeeTypeTestData.DeveloperType.ToDto();
    _employeeTypes = new List<EmployeeType>
        {
           new() { Id = 1, Name = "CAM" },
           new() { Id = 2, Name = "Designer" },
           new() { Id = 3, Name = "Developer" }
        };
    }

Reply:
Agree